packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -935,7 +945,23 @@ class SliverReorderableListState extends State<SliverReorderableList>
 
   void _dragUpdateItems() {
     assert(_dragInfo != null);
-    final double gapExtent = _dragInfo!.itemExtent;
+    double gapExtent = _dragInfo!.itemExtent;
+
+    // For separated lists, the gap should include both the item and its separator
+    if (widget.semanticIndexCallback != null) {
+      // Find separator height and add it to the gap
+      for (final _ReorderableItemState separatorItem in _items.values) {
```

Comment:
SliverReorderableList does not know about separators, and semantic indexes are not exclusive to separators. 
Would this implementation be cleaner if SliverReorderableList had a .separator constructor?

Reply:
Thanks for the response. We cannot merge this PR with the current solution in SliverReorderableList.

> not go deeper into the rabbit hole

We actually call this the yak shave 😉  https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#lazy-programming